src/cmd/build.js

```diff
@@ -60,8 +61,21 @@ function defaultPackageCreator(
           filter: (...args) => fileFilter.wantFile(...args),
         })
         .then((buffer) => {
+          let messageData: any;
+          if (manifestData.default_locale) {
+            messageData = readFileSync(
+               path.join(sourceDir, '_locales', manifestData.default_locale,
+                'messages.json'));
+          }
+          let extensionName: string;
+          if (messageData) {
+            extensionName = JSON.parse(messageData).extensionName.description;
```

Comment:
Also `name` does not always have to be `__MSG_extensionName__` – I, for example, use `__MSG_addon_name__`. Maybe this whole issue would better be handled in when generating [manifestData](https://github.com/mozilla/web-ext/blob/2c06dbd892feb02bd69a0345bee2b4d2d98233b1/src/util/manifest.js)